recipes/nng/all/conanfile.py

```diff
@@ -0,0 +1,81 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from glob import glob
+import os
+
+
+class NngConan(ConanFile):
+    name = "nng"
+    description = "nanomsg-next-generation: light-weight brokerless messaging"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "src/*.cmake", "src/*.txt"]
+    homepage = "https://github.com/nanomsg/nng"
+    license = "MIT"
+    topics = ("nanomsg", "communication", "messaging", "protocols")
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "tests": [True, False],
+        "nngcat": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "tests": False,
+        "nngcat": False
+    }
+
+    _source_subfolder = "source_subfolder"
+    _cmake = None
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+        if self.settings.compiler == "Visual Studio" and \
+                tools.Version(self.settings.compiler.version) < 14:
+            raise ConanInvalidConfiguration("MSVC < 14 is not supported")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["NNG_TESTS"] = self.options.tests
+        self._cmake.definitions["NNG_NNGCAT"] = self.options.nngcat
+        self._cmake.configure(source_folder=self._source_subfolder)
```

Comment:
This will configure the cmake script without calling the cmake helpers (using the `cmake` generator).

You'll need to add the `cmake` generator and add a `CMakeLists.txt` file that adds `nn` as a subdirectory.
Or patch nn's cmake script.

Reply:
Thanks for your comments and hints!

I believe I did as you suggested, but the test package still fails to link nng on Windows (see [this build failure](https://github.com/conan-io/conan-center-index/pull/1887#issuecomment-644010625)). Any ideas? I'm a bit at a loss TBH.

Reply:
I'll look into this when I start my Windows box.